README.md

```diff
@@ -1,16 +1,241 @@
+pictures to help drive these points home. Using sketches will really
+help you "get it." This is why developers in professional settings are
+always sketching their NDS' whiteboards. It really works!
+
+## Identify Nested Structures Can Be Mixed
+
+As we try to more accurately model the world around us, we'll see that it's
+very helpful to mix and match NDS to provide a better model of reality.
+
+Let's update our model of a grid-based vending machine again. Remember,
+they look like this:
+
+![Grid-based vending machine](https://curriculum-content.s3.amazonaws.com/programming-univbasics-5/nested-arrays-lab/vending_grid.png)
+
+We're going to work "top-down" and then re-work the modeling "bottom-up."
+Neither method is more correct than the other. The problem of modeling sometimes
+requires us to be flexible. "Top-down" means: describe the outermost structure
+and then fill it in, and fill in any structures that were just added until you
+reach the smallest structure. "Bottom-up" means: describe the innermost structure
+and "wrap it" with a container repeatedly until the outermost container is reached.
+
+### Array of Array of Array or Hashes...
+
+Ultimately, in order to model a real grid-based vending machine (C20 is a candy bar
+gum, B50 is a drink), we think the right NDS is:
+
+```text
+Array of Array
+...of Array
+...of Hashes
+...with keys `:name` and `:pieces` where
+......:name points to a `String`
+......:pieces points to an Integer count of pieces
+```
+
+We're giving you the answer so that you can think about how this might work for
+yourself. Draw it out on paper and see if you can predict how our explanation
+will work. Teach your friend, your cat, or your imaginary friend Hogarth how ***you***
+think things work **and then** read the explanation below. Don't lose this chance
+to train your brain!
+
+### Explaining the Vending Machine
+
+Many vending machines use a grid to identify which snack one wants. The keyword
+"grid" should immediately suggest an AoA.  Using a coordinate like `[3][2]` we
+might pick the third element on the 4th row (remember, indexes start at 0 in a
+Ruby vending machine).
+
+### Explaining the Coordinate to Array
+
+Here's a picture of what we just described in code:
+
+```ruby
+vending_machine = [
+   [
+     ["Vanilla Cookies", "Pistachio Cookies", "Chocolate Cookies", "Chocolate Chip Cookies"],
+     ["Tooth-Melters", "Tooth-Destroyers", "Enamel Eaters", "Dentist's Nighmare"],
+     ["Gummy Apple", "Gummy Apple", "Gummy Moldy Apple"]
+   ],
+   [
+     ["Grape Drink", "Orange Drink", "Pineapple Drink"],
+     ["Mints", "Curiously Toxic Mints", "US Mints"]
+    ]
+  ]
+#=> [[["Vanilla Cookies", "Pistachio Cookies", "Chocolate Cookies", "Chocolate Chip Cookies"], ["Tooth-Melters", "Tooth-Destroyers", "Enamel Eaters", "Dentist's Nighmare"], ["Gummy Apple", "Gummy Apple", "Gummy Moldy Apple"]], [["Grape Drink", "Orange Drink", "Pineapple Drink"], ["Mints", "Curiously Toxic Mints", "US Mints"]]]
+```
```

Comment:
this is a bit confusing for me, but mainly because of the extra level of nesting without clear explanation - the first three arrays separate from the last two. I think although it is mentioned in the previous section that we're giving the student the answer, I think there needs to be a clearer call out that we're stepping back from AoAoH (currently AoAoAoH) and just doing AoA (currently AoAoA)

Reply:
I cut this. Instead of covering the intermediate topic just went to AoAoH.